pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -163,7 +163,7 @@ func (s *DiscoveryServer) updateModel() {
 		log.Errorf("XDS: failed to update virtual services %v", err)
 	} else {
 		s.virtualServiceConfigs = vservices
-		s.virtualServices = make([]*networking.VirtualService, len(vservices))
+		s.virtualServices = make([]*networking.VirtualService, 0, len(vservices))
```

Comment:
wonder if the change is necessary since the size of the array is fixed (as in len(vservices)).